ruby-challenges.rb

```diff
@@ -10,22 +10,99 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+# def beverage (array, letter)
+#     if array.include?(letter) 
+# p beverages_array
+#     # else
+#     #     if beverages_array.includes?(letter_t) 
+#     #         p beverages_array
+#     end
+# end
+
+
+# p beverage (beverages_array letter_o)
+
+
+# beverages_array.each do |array|
+#     if array.includes? letter
+
+
+def beverage (array, string)
+   new_array = array.select{|word| word.include?(string)}
+    p new_array
+  end
+
+beverage(beverages_array, letter_o)
+
+# def beverage 
+#     if beverages_array == letter_o || letter_t
+#         returns beverages_array
+# end
+# end
+# p beverage
+
+
+# def beverage (array, letter)
+#    if array.select { |word| word.include? 'o' || 't' }
+# end
+# p beverages_array
+
+
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+def states (hash)
+ hash.values.flatten.sort
+end
+
+p states (us_states)
+
+# Output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
+class Bike 
+    attr_accessor :model, :wheels, :age
+    def initialize (model, wheels, current_speed)
+        @model = model
+        @wheels= 2
+        @current_speed= 0
+    end
 
-# Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
+    def get_info
+        "The #{@model} bike has #{@wheels} and is going #{@current_speed}."
+    end
+
+    def accelerate number_param
+        @current_speed += number_param
+      end
+      def decelerate number_param
+            [@current_speed -= number_param, 0].max
```

Comment:
Nice work, I really like this approach to keep the speed from dropping below 0!  Crushing this section 🎉